karabo_data/export.py

```diff
@@ -124,26 +141,61 @@ def _serialize(self, data):
                     main_data[key] = value
 
             msg.extend([
-                pack({'source': src, 'content': 'msgpack'}),
-                pack(main_data)
+                self.pack({
+                    'source': src, 'content': 'msgpack',
+                    'metadata': metadata[src]
+                }),
+                self.pack(main_data)
             ])
             for key, array in arrays:
                 if not array.flags['C_CONTIGUOUS']:
                     array = np.ascontiguousarray(array)
                 msg.extend([
-                    pack({'source': src, 'content': 'array', 'path': key,
-                          'dtype': str(array.dtype), 'shape': array.shape}),
+                    self.pack({
+                        'source': src, 'content': 'array', 'path': key,
+                        'dtype': str(array.dtype), 'shape': array.shape
+                    }),
                     array.data,
                 ])
 
         return msg
 
-    def feed(self, data):
+    def feed(self, data, metadata=None):
         """Push data to the sending queue.
 
         This blocks if the queue already has *maxlen* items waiting to be sent.
+
+        Parameters
+        ----------
+        data : dict
+            Contains train data. The dictionary has to follow the karabo_bridge
+            protocol structure:
+            - keys are source names
+            - values are dict, where the keys are the parameter names and
+              values must be python built-in types or numpy.ndarray.
+        metadata : dict, optional
+            Contains train metadata. The dictionary has to follow the
+            karabo_bridge protocol structure:
+            - keys are (str) source names
+            - values (dict) should contain the following items:
+              {
+                  'timestamp': 1234.567890  # float
+                  'timestamp.sec': '1234'  # str
+                  'timestamp.frac': '567890000000000000'  # str
+                  'timestamp.tid': 1234567890  # int
+              }
+
+            'timestamp' is the Unix epoch
+            'timestamp.sec' is the seconds of Unix epoch
+            'timestamp.frac' is the fractional part of Unix epoch with
```

Comment:
Wording nitpick: the 'Unix epoch' is the zero point of the measurement system. The measurements can be called  'Unix time' or 'seconds since the Unix epoch'. https://en.wikipedia.org/wiki/Unix_time